plugins/middleware/registry.go

```diff
@@ -0,0 +1,15 @@
+package middleware
+
+import "net/http"
+
+type Creator func() Middleware
+
+var Middlewares = map[string]Creator{}
+
+type Middleware interface {
+	Handler() func(http.Handler) http.Handler
+}
+
+func Add(name string, creator Creator) {
+	Middlewares[name] = creator
```

Comment:
Should we log an error or panic when there is already a `creator` set for `name`?

Reply:
Good idea. Probably `panic` makes the most sense to fail ASAP.